x/slashing/tick.go

```diff
@@ -0,0 +1,54 @@
+package slashing
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/abci/types"
+	crypto "github.com/tendermint/go-crypto"
+)
+
+func NewBeginBlocker(sk Keeper) sdk.BeginBlocker {
+	return func(ctx sdk.Context, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+		// Tag the height
+		heightBytes := make([]byte, 8)
+		binary.LittleEndian.PutUint64(heightBytes, uint64(req.Header.Height))
+		tags := sdk.NewTags("height", heightBytes)
+
+		// Deal with any equivocation evidence
+		for _, evidence := range req.ByzantineValidators {
+			var pk crypto.PubKey
+			sk.cdc.MustUnmarshalBinary(evidence.PubKey, &pk)
+			switch {
+			case bytes.Compare(evidence.Type, []byte("doubleSign")) == 0:
```

Comment:
In terms of naming the evidence type or comparing bytes?

Reply:
yeh both. probably we should make type a string ?

Reply:
Maybe. Would we ever want evidence to be parameterized (type could be a binary blob which stored another serialized structure)?

Reply:
This is part of the ABCI spec, seems better not to include an enum in ABCI itself?

Reply:
Oh yeah this could go right in ABCI - `evidence.Type == abci.DoubleSign` I'm not sure why we wouldn't want to do this? I mean is []byte("doublesign") explicitly in abci right now?

Reply:
No, I just made that up - agreed, that sounds great - I just think we should keep the wire type bytes and have some standard constants in ABCI.

Reply:
Now uses evidence type enum - https://github.com/tendermint/abci/pull/252